src/mapillary/interface.py

```diff
@@ -523,10 +559,14 @@ def traffic_signs_in_bbox(
         ...    existed_before='YYYY-MM-DD HH:MM:SS'
         ... )
     """
-
-    return feature.get_map_features_in_bbox_controller(
-        bbox=bbox, filters=filters, filter_values=filter_values, layer="traffic_signs"
-    )
+    if check_bbox_validity(bbox):
+        return feature.get_map_features_in_bbox_controller(
+            bbox=bbox, filters=filters, filter_values=filter_values, layer="traffic_signs"
+        )
+    else:
+        print("Please enter the correct bounding box values")
```

Comment:
Same as above.

Reply:
Done all the above but need a clarity as to how the `bbox_validity_check` function should be called from `interface.py`. An example would be appreciated since I could not find check function calls

Reply:
Ah, my bad.

So for this specific code, and looking at the new additions you made, we can now simply write,

```python
    check_bbox_validity(bbox):
    
    
    return feature.get_map_features_in_bbox_controller(
      bbox=bbox, filters=filters, filter_values=filter_values, layer="traffic_signs"
     )
```

So if the check passes in `check_bbox_validity`, 'no' exceptions will be thrown, hence there will be no need to write an `if` or `else`, which means, no need to return a boolean.

This means we can change the return in `check_bbox_validity` to be `None`, and now we can raise the exception in place of where `False` was being returned.

Because if you don't have a valid bbox, the request itself will be invalid, hence no purpose of making a request to the API endpoint.